### PR TITLE
[v1.13] Backports 2024-03-05

### PR DESCRIPTION
**Pull request by julianwiedmann**
Manual backport of
* [ ] #24107 (partial)
* [ ] #29530
* [ ] #29721
* [ ] #29761
* [ ] #29594
* [ ] #29962
* [ ] #30095

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 29530 29721 29761 29594 29962 30095
```

### CODE REVIEW

**Comment by julianwiedmann** [comment]
/test-backport-1.13

**Comment by julianwiedmann** [comment]
/test-1.25-4.19

**Comment by julianwiedmann** [comment]
/test-1.18-4.19

**Comment by julianwiedmann** [comment]
/test-1.25-4.19

**Review by joestringer — APPROVED** [review]
As @cilium/tophat I can only really defer to you as author that these changes are properly backported. Approve from that perspective.

As a committer I struggle to understand how this fits the [backporting criteria](https://docs.cilium.io/en/v1.15/contributing/release/backports/#backport-criteria-for-x-y-1-z-and-x-y-2-z). Now that we're approaching v1.13.13, users have a pretty high expectation of stability in the releases so it makes me a bit nervous to see such invasive changes. I don't see any test coverage changes in the PR.

**Review by jschwinger233 — APPROVED** [review]
Thank you! The only thing I want to say is about the separate routing rules for v4/v6. It's unique and different from v1.14 backport: [1.13](https://github.com/cilium/cilium/pull/31161/commits/5c0dce0250e4641b1bd3097719f3e17ed88a8666) vs [1.14](https://github.com/cilium/cilium/pull/31160/commits/88c4ef3826095985175560ad9239c6c22125599f)

It would be nice to have some comments for other maintainers.

**Comment by julianwiedmann** [comment]
> Thank you! The only thing I want to say is about the separate routing rules for v4/v6. It's unique and different from v1.14 backport: [1.13](https://github.com/cilium/cilium/pull/31161/commits/5c0dce0250e4641b1bd3097719f3e17ed88a8666) vs [1.14](https://github.com/cilium/cilium/pull/31160/commits/88c4ef3826095985175560ad9239c6c22125599f)
> 
> It would be nice to have some comments for other maintainers.

Right, that one is subtle - thank you! I'll improve the patch description / add a comment.

**Comment by julianwiedmann** [comment]
Updated https://github.com/cilium/cilium/pull/31161/commits/e145638e5dd2f8ff5adc86f35ca214bb998ed261 with a code comment and detailed patch description.

**Comment by julianwiedmann** [comment]
/test-backport-1.13

**Review by jschwinger233 — APPROVED** [review]
Thanks for detailed commit message!

**Pull request merged**
